### PR TITLE
[RAPTOR-8302,8246] split integration tests, update R env to 4.2.1

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
I encountered circular dependency with two tasks: updating R env to R 4 and splitting integration tests, thus this PR has both of them. Better review by commits, there two of them.

1. Splitting tests
    1. DRUM integration tests run in a huge single container that contains all the ML frameworks. This PR splits inference/training tests cases to be run per environment/per task.
    2. Some test cases still run in the same container, but overall tests take significantly less time and tend to be more stable.
        E.g. all the tests running in a single container took 65 minutes. Now the tests cases that still run in that container take 35 minutes. And all the tasks per env are running in parallel and take ~20 minutes.
    3. The ultimate goal is to get rid of the container and run tests native on the node.
    4. Old jenkins jobs  will be removed soon.
2. Updating R environment:
    1. Update R base environment to use official R4.2.1 image
    2. Update R drop environment

### CODE REVIEW

**Review by rvorobii — APPROVED** [review]

File: task_templates/1_transforms/4_r_transform_recipe/custom.R [hunk 1]
```diff
@@ -46,7 +46,8 @@ fit <- function(X, y, output_dir, ...){
     # Drop constant columns
     step_zv(all_predictors()) %>%
     # Numeric preprocessing
-    step_medianimpute(all_numeric()) %>%
+    # step_impute_median in the current example stopped working with R4.2.1
+    # step_impute_median(all_numeric()) %>%
```
[the remark this answers is not in the document]
Comment (eric-s-s) [reply task_templates/1_transforms/4_r_transform_recipe/custom.R:50]: and if this no longer works, can we please just delete it?  same for all the other commented out code

**Comment by eric-s-s** [comment]
i am removing myself as a reviewer for this as i have no context for it.

File: task_templates/1_transforms/4_r_transform_recipe/custom.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (yakov-g) [reply task_templates/1_transforms/4_r_transform_recipe/custom.R]: done

**Review by eric-s-s — APPROVED** [review]
i can approve in a _Very_ limited context.  there are many parts of this PR that i have zero context on.  my approval has been specifically asked for in relation to: https://github.com/datarobot/datarobot-user-models/pull/670/commits/f5f6f760da50c9929bfa744909e0ca25e3626420

and i approve of that part

the rest of this _seems_ good and if it works, then :+1:

**Pull request merged**